tests/main/interfaces-shutdown-introspection/task.yaml

```diff
@@ -0,0 +1,48 @@
+summary: Ensures that introspection of login1 of the shutdown interface works.
+
+systems:
+    # No confinement (AppArmor, Seccomp) available on these systems
+    - -debian-*
+    # unity7 implicit slot (used to access dbus-send), not available on core
+    - -ubuntu-core-*
+
+details: |
+    A snap declaring the shutdown plug is defined, its command just calls
+    the Introspect method on org.freedesktop.login1.
+
+prepare: |
+    echo "Given a snap declaring a plug on the shutdown interface is installed"
+    snapbuild $TESTSLIB/snaps/shutdown-introspection-consumer .
+    snap install --dangerous shutdown-introspection-consumer_1.0_all.snap
+
+restore: |
+    rm -f shutdown-introspection-consumer_1.0_all.snap
+
+execute: |
+    CONNECTED_PATTERN=":shutdown +shutdown-introspection-consumer"
+    DISCONNECTED_PATTERN="(?s).*?\n- +shutdown-introspection-consumer:shutdown"
+
+    echo "Then the plug is shown as disconnected"
+    snap interfaces | grep -Pzq "$DISCONNECTED_PATTERN"
```

Comment:
We are currently using the `MATCH` function instead of grep (still not updated old tests though), replacing all the `grep -Pq[z]` occurrences with `MATCH` should just work.